src/main/java/com/baloise/incubator/argonaut/infrastructure/git/GitDeployPullRequestService.java

```diff
@@ -36,20 +37,22 @@
     @Autowired
     private PullRequestCommentService pullRequestCommentService;
 
-    @Autowired
-    private GitHub gitHub;
+    @Override
+    public void deploy(PullRequest pullRequest) {
+        deployBranch(pullRequest, false);
+    }
 
     @Override
-    public void deploy(PullRequest pullRequest, String deploymentRepoUrl, String newImageTag) {
-        String branchName = "";
-        try {
-            branchName = gitHub.getRepository(pullRequest.getFullName()).getPullRequest(pullRequest.getId()).getHead().getRef();
-        } catch (IOException e) {
-            LOGGER.error("Error getting PR HEAD REF: ", e);
-        }
-        LOGGER.info("Deploying url: {}, pullRequest: {}, newImageTag: {}", deploymentRepoUrl, pullRequest, newImageTag);
-        String sanitizedBranchName = branchName.replace("/", "-");
+    public void promoteToProd(PullRequest pullRequest) {
+        deployBranch(pullRequest, true);
+    }
+
+    private void deployBranch(PullRequest pullRequest, boolean isProd) {
```

Comment:
I dont know if the flag isProd has the right name. Maybe you can invert the boolean and name it useBranchName or branchSpecificEnvironment, . Or let it and name it useMasterBranch.  (or am I thinking to complicated ;-))

Reply:
Its basically a quick hack for the moment. You do not want to deploy to prod from a feature branch i guess. It makes things easier to test for the moment.